From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 8 May 2017 16:14:38 +0200
Subject: [PATCH] {BUG in 4.12-rc1} drivers/media/usb/rainshadow-cec: fix
 -Wstrncat-size

clang reports:

    drivers/media/usb/rainshadow-cec/rainshadow-cec.c:299:22: error: the
    value of the size argument in 'strncat' is too large, might lead to
    a buffer overflow [-Werror,-Wstrncat-size].
                    strncat(cmd, hex, sizeof(cmd));
                                      ^~~~~~~~~~~
    drivers/media/usb/rainshadow-cec/rainshadow-cec.c:299:22: note:
    change the argument to be the free space in the destination buffer
    minus the terminating null byte
                    strncat(cmd, hex, sizeof(cmd));
                                      ^~~~~~~~~~~
                                      sizeof(cmd) - strlen(cmd) - 1
---
 drivers/media/usb/rainshadow-cec/rainshadow-cec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/usb/rainshadow-cec/rainshadow-cec.c b/drivers/media/usb/rainshadow-cec/rainshadow-cec.c
index 541ca543f71f..a6ea9390ad38 100644
--- a/drivers/media/usb/rainshadow-cec/rainshadow-cec.c
+++ b/drivers/media/usb/rainshadow-cec/rainshadow-cec.c
@@ -296,7 +296,7 @@ static int rain_cec_adap_transmit(struct cec_adapter *adap, u8 attempts,
 			 cec_msg_destination(msg), msg->msg[1]);
 		for (i = 2; i < msg->len; i++) {
 			snprintf(hex, sizeof(hex), "%02x", msg->msg[i]);
-			strncat(cmd, hex, sizeof(cmd));
+			strncat(cmd, hex, sizeof(cmd) - strlen(cmd) - 1);
 		}
 	}
 	mutex_lock(&rain->write_lock);
-- 
